operator/controllers/csiscaleoperator_controller.go

```diff
@@ -530,29 +539,71 @@ func (r *CSIScaleOperatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return watchResources[resourceKind][resourceName]
 	}
 
+	restartCSIPodsOnCreateDelete := func(cfgmapData map[string]string) bool {
+		restart := false
+		for key := range cfgmapData {
+			if !strings.HasPrefix(strings.ToUpper(key), "VAR_DRIVER_") {
+				continue
+			}
+			restart = true
+		}
+		return restart
+	}
 	predicateFuncs := func(resourceKind string) predicate.Funcs {
 		return predicate.Funcs{
 			CreateFunc: func(e event.CreateEvent) bool {
 				if isCSIResource(e.Object.GetName(), resourceKind) {
-					r.restartDriverPods(mgr, "created", resourceKind, e.Object.GetName())
+					if resourceKind == corev1.ResourceConfigMaps.String() && e.Object.GetName() == config.CSIEnvVarConfigMap {
+						if restartCSIPodsOnCreateDelete(e.Object.(*corev1.ConfigMap).Data) {
+							r.restartDriverPods(mgr, "created", resourceKind, e.Object.GetName())
+						}
+					} else {
+						r.restartDriverPods(mgr, "created", resourceKind, e.Object.GetName())
+					}
 				} else {
 					return false
 				}
 				return true
 			},
 			UpdateFunc: func(e event.UpdateEvent) bool {
 				if isCSIResource(e.ObjectNew.GetName(), resourceKind) {
-					if !reflect.DeepEqual(e.ObjectOld.(*corev1.Secret).Data, e.ObjectNew.(*corev1.Secret).Data) {
+					if resourceKind == corev1.ResourceSecrets.String() && !reflect.DeepEqual(e.ObjectOld.(*corev1.Secret).Data, e.ObjectNew.(*corev1.Secret).Data) {
 						r.restartDriverPods(mgr, "updated", resourceKind, e.ObjectOld.GetName())
 					}
+
+					if resourceKind == corev1.ResourceConfigMaps.String() {
```

Comment:
Please use `else if` instead of `if`, so that in secret update case, we don't need to check if it is a configmap update again.